platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/CompilationFailedException.java

```diff
@@ -27,8 +29,9 @@
 import java.util.List;
 import java.util.Optional;
 
-public class CompilationFailedException extends RuntimeException implements CompilationFailedIndicator, ProblemAwareFailure {
+public class CompilationFailedException extends RuntimeException implements CompilationFailedIndicator, ProblemAwareFailure, ResolutionProvider {
 
+    public static final String RESOLUTION_MESSAGE = "Check your code and dependencies to fix the compilation error(s)";
```

Comment:
❌ Why are we adding a static resolution in the exception?

This information should come from the `solution` information in the linked problem(s).

Reply:
Correct me if I'm wrong, but the Java compiler diagnostics do not provide any solutions, therefore the `solutions` of the linked problems will be always an empty list.

Reply:
👆 that is correct. We currently can only offer this generic advice.

Reply:
❓ Even if we don't get a specific solution from the compiler itself, can't we add a generic solution to the problem instead of adding that to the exception?